SingularityUI/app/components/taskDetail/TaskDetail.jsx

```diff
@@ -178,9 +178,9 @@ class TaskDetail extends Component {
           file.uiPath = file.name;
         }
 
-        file.fullPath = `${files.fullPathToRoot}/${files.currentDirectory}/${file.name}`;
-        file.downloadLink = `${httpPrefix}://${files.slaveHostname}:${httpPort}/files/download.json?path=${file.fullPath}`;
+        file.fullPath = encodeURIComponent(`${files.fullPathToRoot}/${files.currentDirectory}/${file.name}`);
 
+        file.downloadLink = `${config.apiRoot}/tasks/download/${httpPrefix}/${files.slaveHostname}/port/${httpPort}/path/${file.fullPath}`;
```

Comment:
The value of `httpPrefix` is [based on the Singularity configuration](https://github.com/HubSpot/Singularity/blob/c2d88711176b436a0e1683a83953222b7be0443d/SingularityUI/app/components/taskDetail/TaskDetail.jsx#L168-L173), so we shouldn't need to send this as a parameter to the backend (since the backend also has access to that configuration and knows whether the slave communicates over HTTP or HTTPS and which port it's listening on).